lib/matplotlib/collections.py

```diff
@@ -853,36 +859,78 @@ def get_linewidth(self):
     def get_linestyle(self):
         return self._linestyles
 
+    def _set_mappable_flags(self):
+        """
+        Determine whether edges and/or faces are color-mapped.
+
+        This is a helper for update_scalarmappable.
+        It sets Boolean flags '_edge_is_mapped' and '_face_is_mapped'.
+
+        Returns
+        -------
+        mapping_change: bool
+            True if either flag is True, or if a flag has changed.
+        """
+        edge0 = self._edge_is_mapped
+        face0 = self._face_is_mapped
+        self._edge_is_mapped = False
+        self._face_is_mapped = False
```

Comment:
Later in this function, you check if `edge0 is None`, but these lines will stop that from happening after the first call. Should these lines default to `None`, or is removing `None` the goal here?

Reply:
The latter.  I have added comments to explain the strategy.